src/Models/Location.php

```diff
@@ -0,0 +1,475 @@
+            return "/{$this->market->slug}/faq";
+        }
+
+        return "/{$this->market->slug}/faq/{$this->slug}";
+    }
+
+    public function getBookingsPathAttribute()
+    {
+        if ($this->market->locations_count === 1) {
+            return "/{$this->market->slug}/book-online";
+        }
+
+        return "/{$this->market->slug}/book-online/{$this->slug}";
+    }
+
+    public function getReservationsPathAttribute()
+    {
+        if ($this->market->locations_count === 1) {
+            return "/{$this->market->slug}/reservations";
+        }
+
+        return "/{$this->market->slug}/reservations/{$this->slug}";
+    }
+
+    public function getPhoneLinkAttribute()
+    {
+        return 'tel:' . preg_replace("/[^0-9]/", "", $this->phone);
+    }
+
+    public function getDirectionsUrlAttribute()
+    {
+        return 'https://www.google.com/maps/dir/+Your+location/+' . Str::of($this->title)->replace(' ', '+') . ',@' . $this->latitude . ',' . $this->longitude;
+    }
+
+    public function getBookingsYesterdayAttribute()
+    {
+        return config('locations.model_class.booking')::yesterday()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->count();
+    }
+
+    public function getRevenueBookedYesterdayAttribute()
+    {
+        return number_format((config('locations.model_class.booking')::yesterday()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->sum('amount')
+            +
+                config('locations.model_class.booking')::yesterday()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->sum('total_fees')) / 100, 2);
+    }
+
+    public function getBookingsLastWeekAttribute()
+    {
+        return config('locations.model_class.booking')::week()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->count();
+    }
+
+    public function getRevenueBookedLastWeekAttribute()
+    {
+        return number_format((config('locations.model_class.booking')::week()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->sum('amount')
+            +
+                config('locations.model_class.booking')::week()
+            ->whereHas('order', function (Builder $query) {
+                $query->where('location_id', $this->id);
+            })->sum('total_fees')) / 100, 2);
+    }
+
+    public function stripeEnabled()
+    {
+        return $this->stripe_publishable && $this->stripe_secret;
+    }
+
+    /**
+     * Find location by slot or slot number.
+     *
+     * @param mixed $slot
+     * @return Location|null
+     */
+    public function findBySlot($slot)
```

Comment:
I edited the name of the first parameter. Before it was `$slotNumber`, which was wrong because the rest of the code is using `$slot`, not `$slotNumber`. It means we will get the error on accessing an undefined variable when this check is true

```
if ($slotNumber instanceof Slot) {
  return $slot->room->location;
}
```

Reply:
I think we needed `$slotNumber` due to virtual slots. I added @sl0wik to review this PR to get his feedback on it. Thank you, Phuc.